File: 6.5.2020/AngularDemo2/src/app/employees.service.ts
import { Injectable } from '@angular/core';
import { Employee } from './Employee';

@Injectable( )
export class EmployeesService
{
  getEmployees(): Employee[]
  {
    var emps = [
      new Employee(101, "Scott", "scott@gmail.com"),
      new Employee(103, "Smith", "smith@gmail.com"),
      new Employee(104, "Allen", "allen@gmail.com"),
      new Employee(105, "Ford", "ford@gmail.com"),
      new Employee(106, "John", "john@gmail.com"),
      new Employee(107, "Jones", "jones@gmail.com"),
      new Employee(108, "Jeff", "jeff@gmail.com")
    ];
    return emps;
  }
}

File: 12.5.2020/WebApi/WebApiExample/Services/EmployeesService.cs
using System;
using System.Collections.Generic;
using System.Data.Entity.Core.Metadata.Edm;
using System.Linq;
using System.Web;
using WebApiExample.Models;
using WebApiExample.ServiceContracts;

namespace WebApiExample.Services
{
    public class EmployeesService : IEmployeesService
    {
        public List<Employee> GetEmployees()
        {
            //create object of DbContext class
            using (companyEntities db = new companyEntities())
            {
                //LINQ query to fetch data from table
                List<Employee> emps = db.Employees.ToList();

                db.Configuration.LazyLoadingEnabled = false;
                return emps; //This List<Employee> will be automatically converted into json format (in case of request is sent by angular).
            }
        }

        //Calling StoredProc
        public List<ComplexType1> GetEmpDetails()
        {
            //create object of DbContext class
            companyEntities db = new companyEntities();

            var result = db.usp_GetEmpDetails().ToList();

            return result;
        }
    }
}


